### PR TITLE
Fix Issue in Poller WaitUntil/WaitForCompletion APIs

### PR DESCRIPTION
**Pull request by g2vinay**
Fixes the issue of updating terminal poll context in Poller.
Adds test to fill the testing gap for this scenario.

### CODE REVIEW

File: sdk/core/azure-core/src/test/java/com/azure/core/util/polling/SimpleSyncPollerTests.java [hunk 1]
```diff
@@ -431,6 +432,48 @@ public void testPollerFluxError() throws InterruptedException {
...
+    public void waitUntilShouldPollToCompletion() {
+        final Response activationResponse = new Response("Activated");
+        when(activationOperation.apply(any())).thenReturn(activationResponse);
+
+        LongRunningOperationStatus matchStatus
+            = SUCCESSFULLY_COMPLETED;
+
+        int[] invocationCount = new int[1];
+        invocationCount[0] = -1;
+        //
```
Comment (mssfang) [inline sdk/core/azure-core/src/test/java/com/azure/core/util/polling/SimpleSyncPollerTests.java:445]:
```suggestion
        LongRunningOperationStatus matchStatus = SUCCESSFULLY_COMPLETED;

        int[] invocationCount = new int[1];
        invocationCount[0] = -1;

```

File: sdk/core/azure-core/src/test/java/com/azure/core/util/polling/SimpleSyncPollerTests.java [hunk 2]
```diff
@@ -431,6 +432,48 @@ public void testPollerFluxError() throws InterruptedException {
...
+            switch (invocationCount[0]) {
+                case 0:
+                    return new PollResponse<>(IN_PROGRESS,
+                        new Response("0"), Duration.ofMillis(10));
+                case 1:
+                    return new PollResponse<>(IN_PROGRESS,
+                        new Response("1"), Duration.ofMillis(10));
+                case 2:
+                    return new PollResponse<>(matchStatus,
+                        new Response("1"), Duration.ofMillis(10));
```
Comment (mssfang) [inline sdk/core/azure-core/src/test/java/com/azure/core/util/polling/SimpleSyncPollerTests.java:457]:
```suggestion
                        new Response("2"), Duration.ofMillis(10));
```

Comment (mssfang) [inline sdk/core/azure-core/src/test/java/com/azure/core/util/polling/SimpleSyncPollerTests.java:457]: should be 2?

**Review by mssfang — APPROVED** [review]

File: sdk/core/azure-core/src/test/java/com/azure/core/util/polling/SimpleSyncPollerTests.java (hunk 2 above)
Comment (billwert) [inline sdk/core/azure-core/src/test/java/com/azure/core/util/polling/SimpleSyncPollerTests.java:457]: 2?

**Review by billwert — APPROVED** [review]
one fix but otherwise looks good.

**Pull request merged**